source/opt/eliminate_dead_functions_pass.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "eliminate_dead_functions_pass.h"
+
+#include <unordered_set>
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status EliminateDeadFunctionsPass::Process(ir::Module* module) {
+  bool modified = false;
+
+  // Identify live functions first.  Those that are not live
+  // are dead.
+  std::unordered_set<const ir::Function*> liveFunctionSet;
+  ProcessFunction pfn = [&liveFunctionSet](ir::Function* fp) {
+    liveFunctionSet.insert(fp);
+    return false;
+  };
+  ProcessReachableCallTree(pfn, module);
```

Comment:
Also need to preserve functions with external linkage, and functions reachable from them.

Reply:
I do not understand that this comment is referring to.  The function `ProcessReachableCallTree` will look at functions with the export linkage decoration.  Is there something else?

Reply:
I made this comment before reading the method comment for ProcessReachableCallTree.   I'm withdrawing my complaint. :-)